src/Raven.Server/Documents/Indexes/IndexNameFinder.cs

```diff
@@ -37,13 +37,13 @@ private static string FindName(string[] collections, IReadOnlyCollection<IndexFi
             
             var combinedFields = string.Join("And", fields.Select(x => IndexField.ReplaceInvalidCharactersInFieldName(x.Name)).OrderBy(x => x));
```

Comment:
No, we don't do this based on index names. We compare the query fiels to the fields of existing definition:

https://github.com/ayende/ravendb/blob/v4.0/src/Raven.Server/Documents/Queries/Dynamic/DynamicQueryToIndexMatcher.cs#L102